src/core/utils/url.ts

```diff
@@ -21,14 +21,14 @@ import { ParsedUrlQuery } from 'querystring';
 import { format as formatUrl, parse as parseUrl, UrlObject } from 'url';
 
 export interface URLMeaningfulParts {
-  auth: string | null;
-  hash: string | null;
-  hostname: string | null;
-  pathname: string | null;
-  protocol: string | null;
-  slashes: boolean | null;
-  port: string | null;
-  query: ParsedUrlQuery | {};
+  auth?: string | null;
```

Comment:
@azasypkin what do you think if we relay on node typings and get rid of `null`?
```typescript
export type URLMeaningfulParts = Pick<
  UrlWithParsedQuery,
  'auth' | 'hash' | 'hostname' | 'pathname' | 'protocol' | 'slashes' | 'port' | 'query'
>;
```

Reply:
I'm fine if Node typings are correct. IIRC last time I checked these didn't reflect completely what was happening in reality, like Node typings were saying that `auth/whatever?: string`, but in reality it was `auth: null | string`.

Reply:
Indeed, it works as you described and that's **very** confusing. Let's keep it as is. I will add a comment to re-consider behavior with newer `@types/node` version, because the latest `v10.12.30` has the same problems.

Reply:
We should consider submitting an upstream PR, in my experience it's pretty quick to get fixes merged into the types project.